src/sparseconverter/__init__.py

```diff
@@ -377,67 +376,68 @@ def _build_converters(self):
             }
             for left in SPARSE_COO, SPARSE_GCXS, SPARSE_DOK:
                 for right in SPARSE_COO, SPARSE_GCXS, SPARSE_DOK:
-                    if (left, right) not in self._converters:
-                        self._converters[(left, right)] = partial(
-                            _classes[left].asformat, format=formatcodes[right]
-                        )
+                    self._complete(
+                        (left, right),
+                        partial(_classes[left].asformat, format=formatcodes[right])
+                    )
 
-            for left in NUMPY, CUDA, SCIPY_COO, SCIPY_CSR, SCIPY_CSC:
-                for right in SCIPY_COO, SCIPY_CSR, SCIPY_CSC:
-                    if (left, right) not in self._converters:
-                        self._converters[(left, right)] = chain(_flatsig, _classes[right])
+            for left in (NUMPY, CUDA, SCIPY_COO, SCIPY_CSR, SCIPY_CSC, SCIPY_COO_ARRAY,
+                    SCIPY_CSR_ARRAY, SCIPY_CSC_ARRAY):
+                for right in (SCIPY_COO, SCIPY_CSR, SCIPY_CSC, SCIPY_COO_ARRAY, SCIPY_CSR_ARRAY,
+                        SCIPY_CSC_ARRAY):
+                    self._complete((left, right), chain(_flatsig, _classes[right]))
```

Comment:
For my understanding: we need `_flatsig` because the new array interface does not yet support arbitrary ND arrays, correct?

Reply:
Exactly! No support yet, checked the current version.